### ISSUE TITLE
Simple Runs failing on worker

### ISSUE THREAD

**Issue by Raisins** [codalab/codalab-worksheets#29]
Any cl run command returns failed. 

```
$  cl run date 
0x9d306531d25d4eaa83b56f79474d092c

$  cl info 0x9d306531d25d4eaa83b56f79474d092c -v
bundle_type          : run
uuid                 : 0x9d306531d25d4eaa83b56f79474d092c
data_hash            : 0xf50cf41086b9325bc4195508f2e202c85e08c647
state                : failed
command              : date
owner                : user2(4)
name                 : run-date
created              : 2015-02-17 11:42:12
data_size            : 68
request_cpus         : 0
request_gpus         : 0
exitcode             : 127
job_handle           : 27126
temp_dir             : /Users/Dave/.codalab/temp/0x9d306531d25d4eaa83b56f79474d092c
=== contents ===
name  size
----------
```

From the worker 

```
$  cl worker
2015-02-17 19:41:47: Running worker loop (num_iterations = None, sleep_time = 1)
2015-02-17 19:42:13: 1 CREATED bundles => 1 STAGED, 0 FAILED; 18 bundles still waiting on dependencies.
LocalMachine.start_bundle: copying dependencies of 0x9d306531d25d4eaa83b56f79474d092c to /Users/Dave/.codalab/temp/0x9d306531d25d4eaa83b56f79474d092c
/bin/sh: stdbuf: command not found
work_manager: 0x9d306531d25d4eaa83b56f79474d092c (running): {'job_handle': '27126', 'bundle': RunBundle(uuid='0x9d306531d25d4eaa83b56f79474d092c', name='run-date'), 'success': False, 'exitcode': 127}
Worker.finalize_bundle: installing dependencies to /Users/Dave/.codalab/temp/0x9d306531d25d4eaa83b56f79474d092c (copy=False)
BundleStore.upload: hashing /Users/Dave/.codalab/temp/0x9d306531d25d4eaa83b56f79474d092c
-- END BUNDLE: RunBundle(uuid='0x9d306531d25d4eaa83b56f79474d092c', name='run-date') [failed]
```

Error is command not found but I am able to exicute this command out side of codalab

```
$  date
Tue Feb 17 11:45:30 PST 2015
```

This also fails other commands `python` `sleep`  etc..

**Comment by Raisins** [comment]
Seems to from non linux environments, using command `stdbuf` in local_machine.py

**Comment by percyliang** [comment]
Fixed by #30
